tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/nonjaxrs/valid/ValidLRACSParticipant.java

```diff
@@ -93,14 +95,40 @@ public CompletionStage<Response> complete(URI lraId) {
         });
     }
 
+    /*
+     * The @Status method, if present, MUST report the progress of the compensation. Similarly, if the resource cannot
+     * perform a completion activity immediately.
+     */
     @Status
     public CompletionStage<ParticipantStatus> status(URI lraId) {
         assert lraId != null;
 
         return CompletableFuture.supplyAsync(() -> {
             lraMetricService.incrementMetric(LRAMetricType.Status, lraId, ValidLRACSParticipant.class);
 
-            return ParticipantStatus.Completed;
+            int completing = lraMetricService.getMetric(LRAMetricType.Completing, lraId, ValidLRACSParticipant.class);
+
+            int compensating = lraMetricService.getMetric(LRAMetricType.Compensating, lraId, ValidLRACSParticipant.class);
+
+            int completed = lraMetricService.getMetric(LRAMetricType.Completed, lraId, ValidLRACSParticipant.class);
+
+            int compensated = lraMetricService.getMetric(LRAMetricType.Compensated, lraId, ValidLRACSParticipant.class);
+            // setting as default Completing, but it could also be Active or Compensating
```

Comment:
Now you are setting default to Active.

Reply:
Do you think the default status should be different?

Reply:
Oh, maybe you were referring to the comment. Good point, I have just updated it. Thanks